org.lflang/src/org/lflang/generator/c/CCompiler.java

```diff
@@ -233,8 +235,11 @@ public LFCommand compileCCommand(
      * In C++ mode, the compiler produces .cpp files instead
      * of .c files and uses a C++ compiler to compiler the code.
      */
-    static String getTargetFileName(String fileName, boolean CppMode) {
-        if (CppMode) {
+    static String getTargetFileName(String fileName, boolean CppMode, TargetConfig targetConfig) {
+        if(targetConfig.platform == Platform.ARDUINO) {
+            return fileName + ".ino";
```

Comment:
With our Cmake approach, this would just become the same case as `CppMode`, right?

Reply:
Yes! Converted over to CPPMode with a few tweaks.